packages/sdk/src/wasm/VerifiablePresentationV1/audit-record.ts

```diff
@@ -174,57 +174,67 @@ async function compareContexts(
+    // We expect the order of statements to be identical.
+    requested.forEach((requestedStatement, i) => {
+        const presentationStatement = presentation[i];
+        if (requestedStatement.attributeTag !== presentationStatement.attributeTag) throw atomicError;
 
-        switch (as.type) {
+        switch (requestedStatement.type) {
             case 'AttributeInSet': {
-                if (bs.type !== 'AttributeInSet') throw atomicError;
-                if (as.set.length !== bs.set.length) throw atomicError;
+                if (presentationStatement.type !== 'AttributeInSet') throw atomicError;
+                if (requestedStatement.set.length !== presentationStatement.set.length) throw atomicError;
                 // For the sets, not all implementations used lists where the order items are added
                 // is enforced (i.e. some use Sets).
                 //
                 // This is O(2*n^2), but we don't expect a lot of elements to compare.
                 const setEqual =
-                    as.set.every((asv) => bs.set.some((bsv) => attributeTypeEquals(asv, bsv))) &&
-                    bs.set.every((bsv) => as.set.some((asv) => attributeTypeEquals(asv, bsv)));
+                    requestedStatement.set.every((asv) =>
+                        presentationStatement.set.some((bsv) => attributeTypeEquals(asv, bsv))
+                    ) &&
+                    presentationStatement.set.every((bsv) =>
+                        requestedStatement.set.some((asv) => attributeTypeEquals(asv, bsv))
+                    );
                 if (!setEqual) throw atomicError;
                 break;
             }
             case 'AttributeNotInSet': {
-                if (bs.type !== 'AttributeNotInSet') throw atomicError;
-                if (as.set.length !== bs.set.length) throw atomicError;
+                if (presentationStatement.type !== 'AttributeNotInSet') throw atomicError;
+                if (requestedStatement.set.length !== presentationStatement.set.length) throw atomicError;
 
                 const setEqual =
-                    as.set.every((asv) => bs.set.some((bsv) => attributeTypeEquals(asv, bsv))) &&
-                    bs.set.every((bsv) => as.set.some((asv) => attributeTypeEquals(asv, bsv)));
+                    requestedStatement.set.every((asv) =>
+                        presentationStatement.set.some((bsv) => attributeTypeEquals(asv, bsv))
+                    ) &&
+                    presentationStatement.set.every((bsv) =>
+                        requestedStatement.set.some((asv) => attributeTypeEquals(asv, bsv))
+                    );
                 if (!setEqual) throw atomicError;
                 break;
             }
             case 'RevealAttribute': {
-                if (bs.type !== 'RevealAttribute') throw atomicError;
+                if (presentationStatement.type !== 'AttributeValue') throw atomicError;
```

Comment:
Only this line changed behavior in this function.
The rest is just using proper names for the variables